mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/mail/MaildirMailboxMapper.java

```diff
@@ -264,13 +270,22 @@ public void endRequest() {
      * Stores a copy of a mailbox in a cache valid for one request. This is to enable
      * referring to renamed mailboxes via id.
      * @param mailbox The mailbox to cache
-     * @return The id of the cached mailbox
+     * @return the cached mailbox
      */
     private Mailbox cacheMailbox(Mailbox mailbox) {
-        mailboxCache.add(new SimpleMailbox(mailbox));
-        int id = mailboxCache.size() - 1;
-        ((SimpleMailbox) mailbox).setMailboxId(MaildirId.of(id));
-        return mailbox;
+        boolean isCached = mailboxCache.values().contains(mailbox);
```

Comment:
I'm afraid the `contains()` method is relying on `Mailbox::equals` or in real production is `SimpleMailbox::equals`. I looked at `SimpleMailbox::equals` and found that it use only `MailboxId` comparison. 

That may leads to the consequence `isCached` always is false.

Reply:
Well, before this PR nothing prevent the behaviour you describe from happening...

You are right, however... Concurrent requests will result in potential duplicated entries...

OK, let's use a lock!

Reply:
I will not over engineer this, and will only solve the CI problem with a lock. 

I let the caching issues for others...